libtransmission/crypto-utils-mbedtls.cc

```diff
@@ -165,7 +165,9 @@ void tr_sha256::clear()
 {
     mbedtls_sha256_init(&handle_);
 
-#if MBEDTLS_VERSION_NUMBER >= 0x02070000
+#if MBEDTLS_VERSION_NUMBER >= 0x03000000
+    mbedtls_sha256_starts(&handle_, 0);
+#elif MBEDTLS_VERSION_NUMBER >= 0x02070000
     mbedtls_sha256_starts_ret(&handle_, 0);
 #else
     mbedtls_sha256_starts(&handle_);
```

Comment:
I'm not sure how it was building previously (may be broken since #4495), but `mbedtls_sha256_starts` always took 2 parameters, even in mbedtls 2.0.0:
https://github.com/Mbed-TLS/mbedtls/blob/mbedtls-2.0.0/include/mbedtls/sha256.h#L85

So the whole block should be:
```suggestion
#if MBEDTLS_VERSION_NUMBER < 0x03000000 && MBEDTLS_VERSION_NUMBER >= 0x02070000
     mbedtls_sha256_starts_ret(&handle_, 0);
 #else
     mbedtls_sha256_starts(&handle_, 0);
```